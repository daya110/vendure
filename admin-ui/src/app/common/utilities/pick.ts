/**
 * Returns a new object which is a subject of the input, including only the specified properties.
 * Can be called with a single argument (array of props to pick), in which case it returns a partially
 * applied pick function.
 */
export function pick<T extends object, U extends T = any>(props: Array<keyof T>): (input: U) => T;
export function pick<T extends object, U extends T = any>(input: U, props: Array<keyof T>): T;
export function pick<T extends object, U extends T = any>(
    inputOrProps: U | Array<keyof T>,
    maybeProps?: Array<keyof T>,
): T | ((input: U) => T) {
    if (Array.isArray(inputOrProps)) {
        return (input: U) => _pick(input, inputOrProps);
    } else {
        return _pick(inputOrProps, maybeProps || []);
    }
}

function _pick<T extends object, U extends T = any>(input: U, props: Array<keyof T>): T {
    const output: any = {};
    for (const prop of props) {
        output[prop] = input[prop];
    }
    return output;
}
