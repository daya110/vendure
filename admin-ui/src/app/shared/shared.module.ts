import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { ClarityModule, ClrFormsNextModule } from '@clr/angular';
import { NgSelectModule } from '@ng-select/ng-select';
import { TranslateModule } from '@ngx-translate/core';
import { NgxPaginationModule } from 'ngx-pagination';

import {
    ActionBarComponent,
    ActionBarLeftComponent,
    ActionBarRightComponent,
} from './components/action-bar/action-bar.component';
import { AffixedInputComponent } from './components/affixed-input/affixed-input.component';
import { PercentageSuffixInputComponent } from './components/affixed-input/percentage-suffix-input.component';
import { ChipComponent } from './components/chip/chip.component';
import { CurrencyInputComponent } from './components/currency-input/currency-input.component';
import { CustomFieldControlComponent } from './components/custom-field-control/custom-field-control.component';
import { DataTableColumnComponent } from './components/data-table/data-table-column.component';
import { DataTableComponent } from './components/data-table/data-table.component';
import { FormFieldControlDirective } from './components/form-field/form-field-control.directive';
import { FormFieldComponent } from './components/form-field/form-field.component';
import { FormItemComponent } from './components/form-item/form-item.component';
import { ItemsPerPageControlsComponent } from './components/items-per-page-controls/items-per-page-controls.component';
import { LanguageSelectorComponent } from './components/language-selector/language-selector.component';
import { DialogButtonsDirective } from './components/modal-dialog/dialog-buttons.directive';
import { DialogComponentOutletComponent } from './components/modal-dialog/dialog-component-outlet.component';
import { DialogTitleDirective } from './components/modal-dialog/dialog-title.directive';
import { ModalDialogComponent } from './components/modal-dialog/modal-dialog.component';
import { PaginationControlsComponent } from './components/pagination-controls/pagination-controls.component';
import { SelectToggleComponent } from './components/select-toggle/select-toggle.component';
import { TableRowActionComponent } from './components/table-row-action/table-row-action.component';
import { FileSizePipe } from './pipes/file-size.pipe';
import { ModalService } from './providers/modal/modal.service';

const IMPORTS = [
    ClarityModule,
    ClrFormsNextModule,
    CommonModule,
    FormsModule,
    ReactiveFormsModule,
    RouterModule,
    NgSelectModule,
    NgxPaginationModule,
    TranslateModule,
];

const DECLARATIONS = [
    ActionBarComponent,
    ActionBarLeftComponent,
    ActionBarRightComponent,
    AffixedInputComponent,
    ChipComponent,
    CurrencyInputComponent,
    CustomFieldControlComponent,
    DataTableComponent,
    DataTableColumnComponent,
    ItemsPerPageControlsComponent,
    PaginationControlsComponent,
    TableRowActionComponent,
    FileSizePipe,
    FormFieldComponent,
    FormFieldControlDirective,
    FormItemComponent,
    ModalDialogComponent,
    PercentageSuffixInputComponent,
    DialogComponentOutletComponent,
    DialogButtonsDirective,
    DialogTitleDirective,
    SelectToggleComponent,
    LanguageSelectorComponent,
];

@NgModule({
    imports: IMPORTS,
    exports: [...IMPORTS, ...DECLARATIONS],
    declarations: DECLARATIONS,
    providers: [
        // This needs to be shared, since lazy-loaded
        // modules have their own entryComponents which
        // are unknown to the CoreModule instance of ModalService.
        // See https://github.com/angular/angular/issues/14324#issuecomment-305650763
        ModalService,
    ],
    entryComponents: [ModalDialogComponent],
})
export class SharedModule {}
