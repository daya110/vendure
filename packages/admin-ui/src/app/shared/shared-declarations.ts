export {
    ActionBarComponent,
    ActionBarLeftComponent,
    ActionBarRightComponent,
} from './components/action-bar/action-bar.component';
export { AffixedInputComponent } from './components/affixed-input/affixed-input.component';
export { AssetFileInputComponent } from './components/asset-file-input/asset-file-input.component';
export { AssetGalleryComponent } from './components/asset-gallery/asset-gallery.component';
export { AssetPickerDialogComponent } from './components/asset-picker-dialog/asset-picker-dialog.component';
export { PercentageSuffixInputComponent } from './components/affixed-input/percentage-suffix-input.component';
export {
    ChannelAssignmentControlComponent,
} from './components/channel-assignment-control/channel-assignment-control.component';
export { ChannelBadgeComponent } from './components/channel-badge/channel-badge.component';
export { ChannelLabelPipe } from './pipes/channel-label.pipe';
export { ChipComponent } from './components/chip/chip.component';
export { ConfigurableInputComponent } from './components/configurable-input/configurable-input.component';
export { CurrencyInputComponent } from './components/currency-input/currency-input.component';
export {
    CustomFieldControlComponent,
} from './components/custom-field-control/custom-field-control.component';
export { CustomerLabelComponent } from './components/customer-label/customer-label.component';
export { DataTableColumnComponent } from './components/data-table/data-table-column.component';
export { DataTableComponent } from './components/data-table/data-table.component';
export { DropdownItemDirective } from './components/dropdown/dropdown-item.directive';
export { DropdownMenuComponent } from './components/dropdown/dropdown-menu.component';
export { DropdownTriggerDirective } from './components/dropdown/dropdown-trigger.directive';
export { DropdownComponent } from './components/dropdown/dropdown.component';
export { FacetValueChipComponent } from './components/facet-value-chip/facet-value-chip.component';
export {
    FacetValueSelectorComponent,
} from './components/facet-value-selector/facet-value-selector.component';
export { FormFieldControlDirective } from './components/form-field/form-field-control.directive';
export { FormFieldComponent } from './components/form-field/form-field.component';
export { FormItemComponent } from './components/form-item/form-item.component';
export { FormattedAddressComponent } from './components/formatted-address/formatted-address.component';
export { IfDefaultChannelActiveDirective } from './directives/if-default-channel-active.directive';
export { IfMultichannelDirective } from './directives/if-multichannel.directive';
export { IfPermissionsDirective } from './directives/if-permissions.directive';
export {
    ItemsPerPageControlsComponent,
} from './components/items-per-page-controls/items-per-page-controls.component';
export { LabeledDataComponent } from './components/labeled-data/labeled-data.component';
export { LanguageSelectorComponent } from './components/language-selector/language-selector.component';
export { DialogButtonsDirective } from './components/modal-dialog/dialog-buttons.directive';
export { DialogComponentOutletComponent } from './components/modal-dialog/dialog-component-outlet.component';
export { DialogTitleDirective } from './components/modal-dialog/dialog-title.directive';
export { ExtensionHostComponent } from './components/extension-host/extension-host.component';
export * from './components/extension-host/extension-host-config';
export { ModalDialogComponent } from './components/modal-dialog/modal-dialog.component';
export { ObjectTreeComponent } from './components/object-tree/object-tree.component';
export { OrderStateLabelComponent } from './components/order-state-label/order-state-label.component';
export { PaginationControlsComponent } from './components/pagination-controls/pagination-controls.component';
export { RichTextEditorComponent } from './components/rich-text-editor/rich-text-editor.component';
export { SelectToggleComponent } from './components/select-toggle/select-toggle.component';
export { SimpleDialogComponent } from './components/simple-dialog/simple-dialog.component';
export { TableRowActionComponent } from './components/table-row-action/table-row-action.component';
export { TitleInputComponent } from './components/title-input/title-input.component';
export { EntityInfoComponent } from './components/entity-info/entity-info.component';
export { DatetimePickerComponent } from './components/datetime-picker/datetime-picker.component';
export { CurrencyNamePipe } from './pipes/currency-name.pipe';
export { FileSizePipe } from './pipes/file-size.pipe';
export { SentenceCasePipe } from './pipes/sentence-case.pipe';
export { SortPipe } from './pipes/sort.pipe';
export { StringToColorPipe } from './pipes/string-to-color.pipe';
