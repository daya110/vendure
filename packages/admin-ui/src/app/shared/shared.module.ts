import { OverlayModule } from '@angular/cdk/overlay';
import { CommonModule } from '@angular/common';
import { CUSTOM_ELEMENTS_SCHEMA, NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { ClarityModule } from '@clr/angular';
import { NgSelectModule } from '@ng-select/ng-select';
import { TranslateModule } from '@ngx-translate/core';
import { NgxPaginationModule } from 'ngx-pagination';

import { ActionBarItemsComponent } from './components/action-bar-items/action-bar-items.component';
import {
    ActionBarComponent,
    ActionBarLeftComponent,
    ActionBarRightComponent,
} from './components/action-bar/action-bar.component';
import { DisabledDirective } from './directives/disabled.directive';
import { HasPermissionPipe } from './pipes/has-permission.pipe';
import { ModalService } from './providers/modal/modal.service';
import { CanDeactivateDetailGuard } from './providers/routing/can-deactivate-detail-guard';
import {
    AffixedInputComponent,
    AssetFileInputComponent,
    AssetGalleryComponent,
    AssetPickerDialogComponent,
    ChannelAssignmentControlComponent,
    ChannelBadgeComponent,
    ChannelLabelPipe,
    ChipComponent,
    ConfigurableInputComponent,
    CurrencyInputComponent,
    CurrencyNamePipe,
    CustomerLabelComponent,
    CustomFieldControlComponent,
    CustomFieldLabelPipe,
    DataTableColumnComponent,
    DataTableComponent,
    DatetimePickerComponent,
    DialogButtonsDirective,
    DialogComponentOutletComponent,
    DialogTitleDirective,
    DropdownComponent,
    DropdownItemDirective,
    DropdownMenuComponent,
    DropdownTriggerDirective,
    EntityInfoComponent,
    ExtensionHostComponent,
    FacetValueChipComponent,
    FacetValueSelectorComponent,
    FileSizePipe,
    FormattedAddressComponent,
    FormFieldComponent,
    FormFieldControlDirective,
    FormItemComponent,
    IfDefaultChannelActiveDirective,
    IfMultichannelDirective,
    IfPermissionsDirective,
    ItemsPerPageControlsComponent,
    LabeledDataComponent,
    LanguageSelectorComponent,
    ModalDialogComponent,
    ObjectTreeComponent,
    OrderStateLabelComponent,
    PaginationControlsComponent,
    PercentageSuffixInputComponent,
    RichTextEditorComponent,
    SelectToggleComponent,
    SentenceCasePipe,
    SimpleDialogComponent,
    SortPipe,
    StringToColorPipe,
    TableRowActionComponent,
    TitleInputComponent,
} from './shared-declarations';

const IMPORTS = [
    ClarityModule,
    CommonModule,
    FormsModule,
    ReactiveFormsModule,
    RouterModule,
    NgSelectModule,
    NgxPaginationModule,
    TranslateModule,
    OverlayModule,
];

const DECLARATIONS = [
    ActionBarComponent,
    ActionBarLeftComponent,
    ActionBarRightComponent,
    ConfigurableInputComponent,
    AffixedInputComponent,
    ChipComponent,
    CurrencyInputComponent,
    CurrencyNamePipe,
    CustomerLabelComponent,
    CustomFieldControlComponent,
    DataTableComponent,
    DataTableColumnComponent,
    FacetValueSelectorComponent,
    ItemsPerPageControlsComponent,
    PaginationControlsComponent,
    TableRowActionComponent,
    FacetValueChipComponent,
    FileSizePipe,
    FormFieldComponent,
    FormFieldControlDirective,
    FormItemComponent,
    ModalDialogComponent,
    PercentageSuffixInputComponent,
    DialogComponentOutletComponent,
    DialogButtonsDirective,
    DialogTitleDirective,
    SelectToggleComponent,
    LanguageSelectorComponent,
    RichTextEditorComponent,
    SimpleDialogComponent,
    TitleInputComponent,
    SentenceCasePipe,
    DropdownComponent,
    DropdownMenuComponent,
    SortPipe,
    DropdownTriggerDirective,
    DropdownItemDirective,
    OrderStateLabelComponent,
    FormattedAddressComponent,
    LabeledDataComponent,
    StringToColorPipe,
    ObjectTreeComponent,
    IfPermissionsDirective,
    IfMultichannelDirective,
    HasPermissionPipe,
    ActionBarItemsComponent,
    DisabledDirective,
    AssetFileInputComponent,
    AssetGalleryComponent,
    AssetPickerDialogComponent,
    EntityInfoComponent,
    DatetimePickerComponent,
    ChannelBadgeComponent,
    ChannelAssignmentControlComponent,
    ChannelLabelPipe,
    IfDefaultChannelActiveDirective,
    ExtensionHostComponent,
    CustomFieldLabelPipe,
];

@NgModule({
    imports: IMPORTS,
    exports: [...IMPORTS, ...DECLARATIONS],
    declarations: DECLARATIONS,
    providers: [
        // This needs to be shared, since lazy-loaded
        // modules have their own entryComponents which
        // are unknown to the CoreModule instance of ModalService.
        // See https://github.com/angular/angular/issues/14324#issuecomment-305650763
        ModalService,
        CanDeactivateDetailGuard,
    ],
    entryComponents: [ModalDialogComponent, SimpleDialogComponent, AssetPickerDialogComponent],
    schemas: [CUSTOM_ELEMENTS_SCHEMA],
})
export class SharedModule {}
