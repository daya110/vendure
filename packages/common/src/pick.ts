/**
 * Returns a new object which is a subset of the input, including only the specified properties.
 * Can be called with a single argument (array of props to pick), in which case it returns a partially
 * applied pick function.
 */
export function pick<T extends string>(props: T[]): <U>(input: U) => Pick<U, Extract<keyof U, T>>;
export function pick<U, T extends keyof U>(input: U, props: T[]): { [K in T]: U[K] };
export function pick<U, T extends keyof U>(
    inputOrProps: U | T[],
    maybeProps?: T[],
): { [K in T]: U[K] } | ((input: U) => Pick<U, Extract<keyof U, T>>) {
    if (Array.isArray(inputOrProps)) {
        return (input: U) => _pick(input, inputOrProps);
    } else {
        return _pick(inputOrProps, maybeProps || []);
    }
}

function _pick<U, T extends keyof U>(input: U, props: T[]): { [K in T]: U[K] } {
    const output: any = {};
    for (const prop of props) {
        output[prop] = input[prop];
    }
    return output;
}
