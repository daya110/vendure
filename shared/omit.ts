export type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;

declare const File: any;

/**
 * Type-safe omit function - returns a new object which omits the specified keys.
 */
export function omit<T extends object, K extends keyof T>(obj: T, keysToOmit: K[]): Omit<T, K>;
export function omit<T extends object | any[], K extends keyof T>(obj: T, keysToOmit: string[], recursive: boolean): T;
export function omit<T extends any, K extends keyof T>(obj: T, keysToOmit: string[], recursive: boolean = false): T {
    if ((recursive && !isObject(obj)) || obj instanceof File) {
        return obj;
    }

    if (recursive && Array.isArray(obj)) {
        return obj.map((item: any) => omit(item, keysToOmit, true));
    }

    return Object.keys(obj).reduce((output: any, key) => {
        if (keysToOmit.includes(key)) {
            return output;
        }
        if (recursive) {
            return {...output, [key]: omit((obj as any)[key], keysToOmit, true)};
        }
        return {...output, [key]: (obj as any)[key]};
    }, {} as Omit<T, K>);
}

function isObject(input: any): input is object {
    return typeof input === 'object' && input !== null;
}
