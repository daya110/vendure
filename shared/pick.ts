/**
 * Returns a new object which is a subject of the input, including only the specified properties.
 * Can be called with a single argument (array of props to pick), in which case it returns a partially
 * applied pick function.
 */
export function pick<T extends string>(props: T[]): <U>(input: U) => { [K in T]: K extends keyof U ? U[K]: never };
export function pick<U, T extends keyof U>(input: U, props: T[]): { [K in T]: U[K] };
export function pick<T extends object, U extends T = any>(
    inputOrProps: U | Array<keyof T>,
    maybeProps?: Array<keyof T>,
): T | ((input: U) => T) {
    if (Array.isArray(inputOrProps)) {
        return (input: U) => _pick(input, inputOrProps);
    } else {
        return _pick(inputOrProps, maybeProps || []);
    }
}

function _pick<T extends object, U extends T = any>(input: U, props: Array<keyof T>): T {
    const output: any = {};
    for (const prop of props) {
        output[prop] = input[prop];
    }
    return output;
}
