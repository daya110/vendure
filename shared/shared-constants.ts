/*
 * This file contains constants which are shared between more than one sub-module
 * e.g. values required by both the server and admin-ui.
 */
export const API_PORT = 3000;
export const API_PATH = 'api';
export const AUTH_TOKEN_KEY = 'vendure-auth-token';
export const REFRESH_TOKEN_KEY = 'vendure-refresh-token';
export const DEFAULT_CHANNEL_CODE = '__default_channel__';
export const SUPER_ADMIN_ROLE_CODE = '__super_admin_role__';
export const SUPER_ADMIN_ROLE_DESCRIPTION = 'SuperAdmin';
export const SUPER_ADMIN_USER_IDENTIFIER = 'superadmin';
export const SUPER_ADMIN_USER_PASSWORD = 'superadmin';
export const CUSTOMER_ROLE_CODE = '__customer_role__';
export const CUSTOMER_ROLE_DESCRIPTION = 'Customer';
