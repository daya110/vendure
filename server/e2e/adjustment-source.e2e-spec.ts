import {
    AdjustmentSource,
    AdjustmentType,
    CreateAdjustmentSource,
    GetAdjustmentOperations,
    GetAdjustmentSource,
    GetAdjustmentSourceList,
    UpdateAdjustmentSource,
} from 'shared/generated-types';
import { pick } from 'shared/pick';

import {
    CREATE_ADJUSTMENT_SOURCE,
    GET_ADJUSTMENT_OPERATIONS,
    GET_ADJUSTMENT_SOURCE,
    GET_ADJUSTMENT_SOURCE_LIST,
    UPDATE_ADJUSTMENT_SOURCE,
} from '../../admin-ui/src/app/data/definitions/adjustment-source-definitions';
import {
    AdjustmentActionDefinition,
    AdjustmentConditionDefinition,
} from '../src/config/adjustment/adjustment-types';

import { TestClient } from './test-client';
import { TestServer } from './test-server';

// tslint:disable:no-non-null-assertion

describe('AdjustmentSource resolver', () => {
    const client = new TestClient();
    const server = new TestServer();

    const promoCondition = generateTestCondition('promo_condition', AdjustmentType.PROMOTION);
    const promoCondition2 = generateTestCondition('promo_condition2', AdjustmentType.PROMOTION);
    const taxCondition = generateTestCondition('tax_condition', AdjustmentType.TAX);
    const shippingCondition = generateTestCondition('shipping_condition', AdjustmentType.SHIPPING);

    const promoAction = generateTestAction('promo_action', AdjustmentType.PROMOTION);
    const taxAction = generateTestAction('tax_action', AdjustmentType.TAX);
    const shippingAction = generateTestAction('shipping_action', AdjustmentType.SHIPPING);

    const snapshotProps = ['name', 'type', 'actions', 'conditions', 'enabled'] as Array<
        'name' | 'type' | 'actions' | 'conditions' | 'enabled'
    >;
    let promoAdjustmentSource: AdjustmentSource.Fragment;

    beforeAll(async () => {
        const token = await server.init(
            {
                productCount: 1,
                customerCount: 1,
            },
            {
                adjustmentConditions: [promoCondition, promoCondition2, taxCondition, shippingCondition],
                adjustmentActions: [promoAction, taxAction, shippingAction],
            },
        );
        await client.init();
    }, 60000);

    afterAll(async () => {
        await server.destroy();
    });

    it('createAdjustmentSource promotion', async () => {
        const result = await client.query<CreateAdjustmentSource.Mutation, CreateAdjustmentSource.Variables>(
            CREATE_ADJUSTMENT_SOURCE,
            {
                input: {
                    name: 'promo adjustment source',
                    type: AdjustmentType.PROMOTION,
                    enabled: true,
                    conditions: [
                        {
                            code: promoCondition.code,
                            arguments: ['500'],
                        },
                    ],
                    actions: [
                        {
                            code: promoAction.code,
                            arguments: ['50'],
                        },
                    ],
                },
            },
        );
        promoAdjustmentSource = result.createAdjustmentSource;
        expect(pick(promoAdjustmentSource, snapshotProps)).toMatchSnapshot();
    });

    it('createAdjustmentSource tax', async () => {
        const result = await client.query<CreateAdjustmentSource.Mutation, CreateAdjustmentSource.Variables>(
            CREATE_ADJUSTMENT_SOURCE,
            {
                input: {
                    name: 'tax adjustment source',
                    type: AdjustmentType.TAX,
                    enabled: true,
                    conditions: [
                        {
                            code: taxCondition.code,
                            arguments: ['500'],
                        },
                    ],
                    actions: [
                        {
                            code: taxAction.code,
                            arguments: ['50'],
                        },
                    ],
                },
            },
        );
        expect(pick(result.createAdjustmentSource, snapshotProps)).toMatchSnapshot();
    });

    it('createAdjustmentSource shipping', async () => {
        const result = await client.query<CreateAdjustmentSource.Mutation, CreateAdjustmentSource.Variables>(
            CREATE_ADJUSTMENT_SOURCE,
            {
                input: {
                    name: 'shipping adjustment source',
                    type: AdjustmentType.SHIPPING,
                    enabled: true,
                    conditions: [
                        {
                            code: shippingCondition.code,
                            arguments: ['500'],
                        },
                    ],
                    actions: [
                        {
                            code: shippingAction.code,
                            arguments: ['50'],
                        },
                    ],
                },
            },
        );
        expect(pick(result.createAdjustmentSource, snapshotProps)).toMatchSnapshot();
    });

    it('updateAdjustmentSource', async () => {
        const result = await client.query<UpdateAdjustmentSource.Mutation, UpdateAdjustmentSource.Variables>(
            UPDATE_ADJUSTMENT_SOURCE,
            {
                input: {
                    id: promoAdjustmentSource.id,
                    conditions: [
                        {
                            code: promoCondition.code,
                            arguments: ['90'],
                        },
                        {
                            code: promoCondition2.code,
                            arguments: ['10'],
                        },
                    ],
                },
            },
        );
        expect(pick(result.updateAdjustmentSource, snapshotProps)).toMatchSnapshot();
    });

    it('adjustmentSource', async () => {
        const result = await client.query<GetAdjustmentSource.Query, GetAdjustmentSource.Variables>(
            GET_ADJUSTMENT_SOURCE,
            {
                id: promoAdjustmentSource.id,
            },
        );

        expect(result.adjustmentSource!.name).toBe(promoAdjustmentSource.name);
    });

    it('adjustmentSources, type = promotion', async () => {
        const result = await client.query<GetAdjustmentSourceList.Query, GetAdjustmentSourceList.Variables>(
            GET_ADJUSTMENT_SOURCE_LIST,
            {
                type: AdjustmentType.PROMOTION,
            },
        );

        expect(result.adjustmentSources.totalItems).toBe(1);
        expect(result.adjustmentSources.items[0].name).toBe('promo adjustment source');
    });

    it('adjustmentSources, type = tax', async () => {
        const result = await client.query<GetAdjustmentSourceList.Query, GetAdjustmentSourceList.Variables>(
            GET_ADJUSTMENT_SOURCE_LIST,
            {
                type: AdjustmentType.TAX,
            },
        );

        expect(result.adjustmentSources.totalItems).toBe(1);
        expect(result.adjustmentSources.items[0].name).toBe('tax adjustment source');
    });

    it('adjustmentSources, type = shipping', async () => {
        const result = await client.query<GetAdjustmentSourceList.Query, GetAdjustmentSourceList.Variables>(
            GET_ADJUSTMENT_SOURCE_LIST,
            {
                type: AdjustmentType.SHIPPING,
            },
        );

        expect(result.adjustmentSources.totalItems).toBe(1);
        expect(result.adjustmentSources.items[0].name).toBe('shipping adjustment source');
    });

    it('adjustmentOperations, type = promotion', async () => {
        const result = await client.query<GetAdjustmentOperations.Query, GetAdjustmentOperations.Variables>(
            GET_ADJUSTMENT_OPERATIONS,
            {
                type: AdjustmentType.PROMOTION,
            },
        );

        expect(result.adjustmentOperations).toMatchSnapshot();
    });

    it('adjustmentOperations, type = tax', async () => {
        const result = await client.query<GetAdjustmentOperations.Query, GetAdjustmentOperations.Variables>(
            GET_ADJUSTMENT_OPERATIONS,
            {
                type: AdjustmentType.TAX,
            },
        );

        expect(result.adjustmentOperations).toMatchSnapshot();
    });

    it('adjustmentOperations, type = shipping', async () => {
        const result = await client.query<GetAdjustmentOperations.Query, GetAdjustmentOperations.Variables>(
            GET_ADJUSTMENT_OPERATIONS,
            {
                type: AdjustmentType.SHIPPING,
            },
        );

        expect(result.adjustmentOperations).toMatchSnapshot();
    });
});

function generateTestCondition(code: string, type: AdjustmentType): AdjustmentConditionDefinition {
    return {
        code,
        description: `description for ${code}`,
        args: [{ name: 'arg', type: 'money' }],
        type,
        predicate: (order, args) => true,
    };
}

function generateTestAction(code: string, type: AdjustmentType): AdjustmentActionDefinition {
    return {
        code,
        description: `description for ${code}`,
        args: [{ name: 'percentage', type: 'percentage' }],
        type,
        calculate: (order, args) => {
            return [{ amount: 42 }];
        },
    };
}
