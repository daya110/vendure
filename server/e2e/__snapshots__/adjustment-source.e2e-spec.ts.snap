// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdjustmentSource resolver adjustmentOperations, type = promotion 1`] = `
Object {
  "actions": Array [
    Object {
      "args": Array [
        Object {
          "name": "percentage",
          "type": "percentage",
          "value": null,
        },
      ],
      "code": "promo_action",
      "description": "description for promo_action",
      "type": "PROMOTION",
    },
  ],
  "conditions": Array [
    Object {
      "args": Array [
        Object {
          "name": "arg",
          "type": "money",
          "value": null,
        },
      ],
      "code": "promo_condition",
      "description": "description for promo_condition",
      "type": "PROMOTION",
    },
    Object {
      "args": Array [
        Object {
          "name": "arg",
          "type": "money",
          "value": null,
        },
      ],
      "code": "promo_condition2",
      "description": "description for promo_condition2",
      "type": "PROMOTION",
    },
  ],
}
`;

exports[`AdjustmentSource resolver adjustmentOperations, type = shipping 1`] = `
Object {
  "actions": Array [
    Object {
      "args": Array [
        Object {
          "name": "percentage",
          "type": "percentage",
          "value": null,
        },
      ],
      "code": "shipping_action",
      "description": "description for shipping_action",
      "type": "SHIPPING",
    },
  ],
  "conditions": Array [
    Object {
      "args": Array [
        Object {
          "name": "arg",
          "type": "money",
          "value": null,
        },
      ],
      "code": "shipping_condition",
      "description": "description for shipping_condition",
      "type": "SHIPPING",
    },
  ],
}
`;

exports[`AdjustmentSource resolver adjustmentOperations, type = tax 1`] = `
Object {
  "actions": Array [
    Object {
      "args": Array [
        Object {
          "name": "percentage",
          "type": "percentage",
          "value": null,
        },
      ],
      "code": "tax_action",
      "description": "description for tax_action",
      "type": "TAX",
    },
  ],
  "conditions": Array [
    Object {
      "args": Array [
        Object {
          "name": "arg",
          "type": "money",
          "value": null,
        },
      ],
      "code": "tax_condition",
      "description": "description for tax_condition",
      "type": "TAX",
    },
  ],
}
`;

exports[`AdjustmentSource resolver createAdjustmentSource promotion 1`] = `
Object {
  "actions": Array [
    Object {
      "args": Array [
        Object {
          "name": "percentage",
          "type": "percentage",
          "value": "50",
        },
      ],
      "code": "promo_action",
      "description": "description for promo_action",
      "type": "PROMOTION",
    },
  ],
  "conditions": Array [
    Object {
      "args": Array [
        Object {
          "name": "arg",
          "type": "money",
          "value": "500",
        },
      ],
      "code": "promo_condition",
      "description": "description for promo_condition",
      "type": "PROMOTION",
    },
  ],
  "enabled": true,
  "name": "promo adjustment source",
  "type": "PROMOTION",
}
`;

exports[`AdjustmentSource resolver createAdjustmentSource shipping 1`] = `
Object {
  "actions": Array [
    Object {
      "args": Array [
        Object {
          "name": "percentage",
          "type": "percentage",
          "value": "50",
        },
      ],
      "code": "shipping_action",
      "description": "description for shipping_action",
      "type": "SHIPPING",
    },
  ],
  "conditions": Array [
    Object {
      "args": Array [
        Object {
          "name": "arg",
          "type": "money",
          "value": "500",
        },
      ],
      "code": "shipping_condition",
      "description": "description for shipping_condition",
      "type": "SHIPPING",
    },
  ],
  "enabled": true,
  "name": "shipping adjustment source",
  "type": "SHIPPING",
}
`;

exports[`AdjustmentSource resolver createAdjustmentSource tax 1`] = `
Object {
  "actions": Array [
    Object {
      "args": Array [
        Object {
          "name": "percentage",
          "type": "percentage",
          "value": "50",
        },
      ],
      "code": "tax_action",
      "description": "description for tax_action",
      "type": "TAX",
    },
  ],
  "conditions": Array [
    Object {
      "args": Array [
        Object {
          "name": "arg",
          "type": "money",
          "value": "500",
        },
      ],
      "code": "tax_condition",
      "description": "description for tax_condition",
      "type": "TAX",
    },
  ],
  "enabled": true,
  "name": "tax adjustment source",
  "type": "TAX",
}
`;

exports[`AdjustmentSource resolver updateAdjustmentSource 1`] = `
Object {
  "actions": Array [
    Object {
      "args": Array [
        Object {
          "name": "percentage",
          "type": "percentage",
          "value": "50",
        },
      ],
      "code": "promo_action",
      "description": "description for promo_action",
      "type": "PROMOTION",
    },
  ],
  "conditions": Array [
    Object {
      "args": Array [
        Object {
          "name": "arg",
          "type": "money",
          "value": "90",
        },
      ],
      "code": "promo_condition",
      "description": "description for promo_condition",
      "type": "PROMOTION",
    },
    Object {
      "args": Array [
        Object {
          "name": "arg",
          "type": "money",
          "value": "10",
        },
      ],
      "code": "promo_condition2",
      "description": "description for promo_condition2",
      "type": "PROMOTION",
    },
  ],
  "enabled": true,
  "name": "promo adjustment source",
  "type": "PROMOTION",
}
`;
